src/util/kaldi-table-inl.h

```diff
@@ -302,6 +325,24 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
       if (!key_.empty() && !data_rxfilename_.empty()) {
```

Comment:
The use of data_rxfilename_ to store the [rxfilename+range] at one point and [rxfilename] at another is rather inelegant.  I think you should use a temporary local variable to store the rest-of-line,  determine whether or not it contains a range, and only then set the class member variables.
And I don't think the way you store the previous data_rxfilename_ is very good either.  The only time you care about this is to test if it's the same as the current one.  What would be better I think is to split the state up further: have states
kHaveScpLine, kObjectLoaded, kObjectLoadFailed, and then to only be used when you have a range: states kRangeExtracted, kRangeExtractionFailed.
Then inside NextScpLine, once you work out what the next data_rxfilename_ would be, if the previous state was one of kObjectLoaded or kRangeExtracted or kRangeExtractionFailed, you know you have the object, so if the rxfilename is the same you can just set the state to kObjectLoaded.
This is more in the spirit of the original design.
It may (or may not) make sense to have a separate function to extract the range.